File: src/main/java/io/github/chaosdave34/kitpvp/abilities/impl/archer/LeapAbility.java
package io.github.chaosdave34.kitpvp.abilities.impl.archer;

import io.github.chaosdave34.kitpvp.abilities.Ability;
import io.github.chaosdave34.kitpvp.abilities.AbilityType;
import net.kyori.adventure.text.Component;
import org.bukkit.entity.Player;
import org.bukkit.util.Vector;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public class LeapAbility extends Ability {
    public LeapAbility() {
        super("leap", "Leap", AbilityType.RIGHT_CLICK, 20);
    }

    @Override
    public @NotNull List<Component> getDescription() {
        return createSimpleDescription(
                "Leap into the direction",
                "you are looking at."
        );
    }

    @Override
    public boolean onAbility(Player p) {
        Vector launchVector = p.getEyeLocation().getDirection().normalize();
        launchVector.multiply(2);
        p.setVelocity(launchVector);
        return true;
    }
}

File: src/main/kotlin/io/github/chaosdave34/kitpvp/abilities/impl/archer/LeapAbility.kt
package io.github.chaosdave34.kitpvp.abilities.impl.archer

import io.github.chaosdave34.kitpvp.abilities.Ability
import net.kyori.adventure.text.Component
import org.bukkit.entity.Player

class LeapAbility : Ability("leap", "Leap", Type.RIGHT_CLICK, 10) {
    override fun getDescription(): List<Component> {
        return createSimpleDescription(
            "Leap into the direction",
            "you are looking at."
        )
    }

    override fun onAbility(player: Player): Boolean {
        val launchVector = player.eyeLocation.direction.normalize()
        launchVector.multiply(2)
        player.velocity = launchVector
        return true
    }
}